astroquery/esa/xmm_newton/core.py

```
@@ -266,5 +268,159 @@ def get_columns(self, table_name, *, only_names=True, verbose=False):
+                    log.warning("Invalid band %u" % i)
+                    band.remove(i)
+
+        if instrument == []:
+            instrument = _instrument
+        else:
+            for i in instrument:
+                if i not in _instrument:
+                    log.warning("Invalid instrument %s" % i)
+                    instrument.remove(i)
+        try:
+            with tarfile.open(filename, "r") as tar:
+                for i in tar.getmembers():
+                    paths = os.path.split(i.name)
+                    fname = paths[1]
+                    paths = os.path.split(paths[0])
+                    if paths[1] != "pps":
+                        continue
+                    fname_info = self._parse_filename(fname)
+                    if fname_info["X"] != "P":
+                        continue
+                    if not fname_info["I"] in instrument:
+                        continue
+                    if not int(fname_info["S"]) in band:
+                        continue
+                    if not fname_info["T"] in _product_type:
+                        continue
+                    tar.extract(i, _path)
+                    if not ret.get(int(fname_info["S"])):
+                        ret[int(fname_info["S"])] = {}
+                    if fname_info["T"] == "DETMSK":
+                        ret[int(fname_info["S"])][fname_info["I"] + "_det"] =\
+                            os.path.abspath(os.path.join(_path, i.name))
+                    elif fname_info["T"] == "EXPMAP":
+                        ret[int(fname_info["S"])][fname_info["I"] + "_expo"] =\
+                            os.path.abspath(os.path.join(_path, i.name))
+                    else:
+                        ret[int(fname_info["S"])][fname_info["I"]] =\
+                            os.path.abspath(os.path.join(_path, i.name))
+        except FileNotFoundError:
+            log.error("File %s not found" % (filename))
+            return {}
```

This should just raise an exception.